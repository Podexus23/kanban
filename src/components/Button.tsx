import styles from "./Button.module.css";

const Button = ({ name, size, onClick, title = "" }) => {
  return (
    <button
      title={title}
      onClick={onClick}
      className={`${styles[size]} ${styles.button}`}
    >
      {name}
    </button>
  );
};

export default Button;
