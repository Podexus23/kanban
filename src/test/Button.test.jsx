import { render, screen } from "@testing-library/react";
import Button from "../components/Button";
import userEvent from "@testing-library/user-event";
import { describe, expect, test, vi } from "vitest";

describe("button tests", () => {
  test("matches snapshot", () => {
    const { container } = render(
      <Button
        name={"my name"}
        title="some title"
        onClick={() => console.log("test btn")}
        size={"medium"}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders button with text and handles click", async () => {
    const user = userEvent.setup();
    const handleClick = vi.fn(); // Mock-функция

    render(<Button name={"click me"} onClick={handleClick} />);

    // Проверка рендеринга
    const button = screen.getByRole("button", { name: /click me/i });
    expect(button).toBeInTheDocument();

    // Тест клика
    await user.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
