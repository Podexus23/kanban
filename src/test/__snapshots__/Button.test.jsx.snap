// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button tests > matches snapshot 1`] = `
<div>
  <button
    class="_medium_29f8d0 _button_29f8d0"
    title="some title"
  >
    my name
  </button>
</div>
`;

exports[`matches snapshot 1`] = `
<div>
  <button
    class="_medium_29f8d0 _button_29f8d0"
    title="some title"
  >
    my name
  </button>
</div>
`;
